new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java

```diff
@@ -214,156 +224,281 @@ public void testCreateMiiDomain() {
-    encryptionSecretMap.put("username", "weblogicenc");
-    encryptionSecretMap.put("password", "weblogicenc");
-    secretCreated = assertDoesNotThrow(() -> createSecret(new V1Secret()
-        .metadata(new V1ObjectMeta()
-            .name(encryptionSecretName)
-            .namespace(domainNamespace))
-        .stringData(encryptionSecretMap)), "Create secret failed with ApiException");
-    assertTrue(secretCreated, String.format("create secret failed for %s", encryptionSecretName));
+    assertDoesNotThrow(() -> createDomainSecret(encryptionSecretName, "weblogicenc",
+            "weblogicenc", domainNamespace),
+             String.format("createSecret failed for %s", encryptionSecretName));
 
     // create the domain CR
-    Domain domain = new Domain()
-        .apiVersion(API_VERSION)
-        .kind("Domain")
-        .metadata(new V1ObjectMeta()
-            .name(domainUid)
-            .namespace(domainNamespace))
-        .spec(new DomainSpec()
-            .domainUid(domainUid)
-            .domainHomeSourceType("FromModel")
-            .image(miiImage)
-            .addImagePullSecretsItem(new V1LocalObjectReference()
-                .name(repoSecretName))
-            .webLogicCredentialsSecret(new V1SecretReference()
-                .name(adminSecretName)
-                .namespace(domainNamespace))
-            .includeServerOutInPodLog(true)
-            .serverStartPolicy("IF_NEEDED")
-            .serverPod(new ServerPod()
-                .addEnvItem(new V1EnvVar()
-                    .name("JAVA_OPTIONS")
-                    .value("-Dweblogic.StdoutDebugEnabled=false"))
-                .addEnvItem(new V1EnvVar()
-                    .name("USER_MEM_ARGS")
-                    .value("-Djava.security.egd=file:/dev/./urandom ")))
-            .adminServer(new AdminServer()
-                .serverStartState("RUNNING")
-                .adminService(new AdminService()
-                    .addChannelsItem(new Channel()
-                        .channelName("default")
-                        .nodePort(30711))))
-            .addClustersItem(new Cluster()
-                .clusterName("cluster-1")
-                .replicas(replicaCount)
-                .serverStartState("RUNNING"))
-            .configuration(new Configuration()
-                .model(new Model()
-                    .domainType("WLS")
-                    .runtimeEncryptionSecret(encryptionSecretName))));
-
-    logger.info("Create domain custom resource for domainUID {0} in namespace {1}",
-        domainUid, domainNamespace);
-    assertTrue(assertDoesNotThrow(() -> createDomainCustomResource(domain),
-        String.format("Create domain custom resource failed with ApiException for %s in namespace %s",
-            domainUid, domainNamespace)),
-        String.format("Create domain custom resource failed with ApiException for %s in namespace %s",
-            domainUid, domainNamespace));
-
+    int adminNodePort = 30711;
```

Comment:
not for this PR - but in general - we probably need a reliable way to get a free node port

Reply:
Anil mentioned that he sets "0" to adminNodePort and it picks the available port. @markxnelson Is it good idea to do that? It may not be in the node port range.

Reply:
This is actually a good idea.  The Kubernetes doc says that if you don't specify a port for a NodePort type service then Kubernetes itself will select an available port.  It looks like using "0" is an undocumented way to get the same behavior.  After the NodePort is created, you'll need to look at the service to find out what port was chosen.

We've not done anything with this more generally because many of our nodeport use cases, i.e. T3, require the port in WebLogic to be the same as the external port and so this late binding would be difficult.

Reply:
Dongbo and I discussed it, but I decided against using 0 just yet. Wanted to see what others thought about it. I could find a available free port in a range, but what should the range be?